pkg/reader/reader.go

```diff
@@ -99,13 +97,28 @@ func (r *FileReader) Read() error {
 			numErrorLines++
 		}
 
-		if r.StopFlag || (r.File.Limit != nil && *r.File.Limit > 0 && *r.File.Limit <= lineNum) {
+		if r.StopFlag || (r.File.Limit != nil && *r.File.Limit > 0 && int64(*r.File.Limit) <= lineNum) {
 			break
 		}
 	}
 
-	r.BatchMgr.Done()
-	logger.Infof("Total lines of file(%s) is: %d, error lines: %d", *r.File.Path, lineNum, numErrorLines)
+	return
+}
+
+func (r *FileReader) Read() error {
+	var lineNumTotal int64
+	var numErrorLinesTotal int64
```

Comment:
You can check whether `r.File.Path` is a directory. If true, list all files and read contents of each file. otherwise, handle the data file same as before.